Release notes for v1.1.0 (#4860)
grafana/k6: release notes/v1.1.0.md

- [#4831](https://github.com/grafana/k6/pull/4831) Fixes the default user defined in the Dockerfile.
- [#4833](https://github.com/grafana/k6/pull/4833) Retrieves secrets from Vault and adjusts the 'k6packager' deployment process.
- [#4837](https://github.com/grafana/k6/pull/4837), [#4847](https://github.com/grafana/k6/pull/4847) Prevent Codecov from running on forks.
- [#4848](https://github.com/grafana/k6/pull/4848) Enables CI pipelines to be executed also on ARM (`ubuntu-24.04-arm`). Thanks, @nadiamoe, for the contribution!
- [#4855](https://github.com/grafana/k6/pull/4855) Adds some changes to make Browser tests more stable.
- [#4780](https://github.com/grafana/k6/pull/4780), [#4781](https://github.com/grafana/k6/pull/4781), [#4782](https://github.com/grafana/k6/pull/4782), [#4786](https://github.com/grafana/k6/pull/4786), [#4798](https://github.com/grafana/k6/pull/4798), [#4816](https://github.com/grafana/k6/pull/4816), [#4821](https://github.com/grafana/k6/pull/4821), [#4835](https://github.com/grafana/k6/pull/4835), [#4840](https://github.com/grafana/k6/pull/4840), [#4859](https://github.com/grafana/k6/pull/4859) Update direct dependencies.

## _Optional_ Roadmap

> _Discussion of future plans_

Author: Something you would like to add here? cc/ @grafana/k6-core

Reviewer: Maybe @oleiade 's functional testing ?

Reviewer 2: Added some elements, thanks for the heads-up folks 🙇🏻

k6 `v1.1.0` is here 🎉! This release includes:

- New `count`, `nth`, `first` and `last` methods on the `Locator` API [#4797](https://github.com/grafana/k6/pull/4797), [#4825](https://github.com/grafana/k6/pull/4825)

> - The `k6/experimental/webcrypto` module is now globally available.

Reviewer: ```suggestion
- The `k6/experimental/webcrypto` module is removed as its functionality is globally available.

- Group results in the `full` end-of-test summary are now sorted as in code and properly indented.

## Breaking changes

> ### Remove experimental `k6/experimental/webcrypto` module [#4851](https://github.com/grafana/k6/pull/4851)

Author: I'd prefer to leave it there in order to give it proper visibility, as anybody still using the experimental import path will encounter this.

Could you suggest a sentence to clarify what you said? I'll be happy to add it as an explanation next to the main title.

> The `k6/experimental/webcrypto` module is now globally available, and the experimental import is no longer available.

Reviewer: ```suggestion
The [WebCrypto API](https://grafana.com/docs/k6/latest/javascript-api/crypto/) has been globally available since v1.0.0-rc1 (or v0.58.0), and now the  experimental import (`k6/experimental/webcrypto`) is no longer available.

The required change for users should be to just remove the `import` and the rest of the code should just work. 
```

- [#4785](https://github.com/grafana/k6/pull/4785) Updates the `chromedp/cdproto` dependency and adjusts the Browser module accordingly.
- [#4800](https://github.com/grafana/k6/pull/4800) Allows un/marshaling of invalid UTF-8 when using JSONv2 within the Browser module. 
- [#4802](https://github.com/grafana/k6/pull/4802) Fixes the `examples/grpc_server` and updates its dependencies. 
- [#4822](https://github.com/grafana/k6/pull/4822) Uses the `CODECOV_TOKEN` variable for GH Workflows from Vault.
- [#4831](https://github.com/grafana/k6/pull/4831) Fixes the default user defined in the Dockerfile.
- [#4833](https://github.com/grafana/k6/pull/4833) Retrieves secrets from Vault and adjusts the 'k6packager' deployment process.
- [#4837](https://github.com/grafana/k6/pull/4837), [#4847](https://github.com/grafana/k6/pull/4847) Prevent Codecov from running on forks.
- [#4848](https://github.com/grafana/k6/pull/4848) Enables CI pipelines to be executed also on ARM (`ubuntu-24.04-arm`). Thanks, @nadiamoe, for the contribution!
- [#4855](https://github.com/grafana/k6/pull/4855) Adds some changes to make Browser tests more stable.

> - [#4780](https://github.com/grafana/k6/pull/4780), [#4781](https://github.com/grafana/k6/pull/4781), [#4782](https://github.com/grafana/k6/pull/4782), [#4786](https://github.com/grafana/k6/pull/4786), [#4798](https://github.com/grafana/k6/pull/4798), [#4816](https://github.com/grafana/k6/pull/4816), [#4821](https://github.com/grafana/k6/pull/4821), [#4835](https://github.com/grafana/k6/pull/4835), [#4840](https://github.com/grafana/k6/pull/4840), [#4859](https://github.com/grafana/k6/pull/4859) Update direct dependencies.

Reviewer: ```suggestion
- [#4780](https://github.com/grafana/k6/pull/4780), [#4781](https://github.com/grafana/k6/pull/4781), [#4782](https://github.com/grafana/k6/pull/4782), [#4786](https://github.com/grafana/k6/pull/4786), [#4798](https://github.com/grafana/k6/pull/4798), [#4816](https://github.com/grafana/k6/pull/4816), [#4821](https://github.com/grafana/k6/pull/4821), [#4835](https://github.com/grafana/k6/pull/4835), [#4840](https://github.com/grafana/k6/pull/4840) Update direct dependencies.
```

This is the Sobek/goja update, but it was reverted. I do not think we need to explain this in teh release notes though
